File: MomentumiOS/MomentumiOSTests/Transactions/FakeTransactionController.swift
//
//  FakeTransactionController.swift
//  MomentumiOSTests
//
//  Created by Mwai Banda on 8/8/22.
//  Copyright © 2022 Momentum. All rights reserved.
//

@testable import MomentumSDK

class FakeTransactionController: TransactionController {
    var transactions = [MomentumTransaction]()
    func addTransaction(description: String, date: String, amount: Double, isSeen: Bool) {
        transactions.append(MomentumTransaction(id: Int64(transactions.count), description: description, date: date, amount: amount, is_seen: isSeen))
    }
    
    func deleteAllTransactions() {
        transactions.removeAll()
    }
    
    func deleteTransactionById(transactionId: Int32) {
        let index = transactions.firstIndex(where: { $0.id == transactionId })!
        transactions.remove(at: index)
    }
    
    func getAllTransactions(onCompletion: @escaping ([MomentumTransaction]) -> Void) {
        onCompletion(transactions)
    }
    
}

File: MomentumAndroid/src/test/java/com/mwaibanda/momentum/android/presentation/transaction/FakeTransactionController.kt
package com.mwaibanda.momentum.android.presentation.transaction

import com.mwaibanda.momentum.data.db.MomentumTransaction
import com.mwaibanda.momentum.domain.controller.TransactionController
import com.mwaibanda.momentum.domain.models.Transaction
import com.mwaibanda.momentum.utils.DataResponse

class FakeTransactionController: TransactionController {
    private var transactions = mutableListOf<MomentumTransaction>()
    override fun addTransactions(transactions: List<MomentumTransaction>) {
        this.transactions.addAll(transactions)
    }

    override fun postTransactionInfo(
        transaction: Transaction,
        onCompletion: (DataResponse<Int>) -> Unit,
    ) {
        transactions.add(transaction.toMomentumTransaction())
    }

    override fun getTransactions(
        userId: String,
        onCompletion: (DataResponse<List<Transaction>>) -> Unit,
    ) {
        onCompletion(DataResponse.Success(emptyList()))
    }

    override fun addTransaction(
        description: String,
        date: String,
        amount: Double,
        isSeen: Boolean
    ) {
        transactions.add(
            MomentumTransaction(
                id = (transactions.count() + 1).toLong(),
                description = description,
                date = date,
                amount = amount,
                is_seen = isSeen
            )
        )
    }


    override fun getMomentumTransactions(onCompletion: (List<MomentumTransaction>) -> Unit) {
        onCompletion(transactions)
    }

    override fun deleteAllTransactions() {
        transactions.clear()
    }

    override fun deleteTransactionById(transactionId: Int) {
        val index = transactions.indexOfFirst { it.id == transactionId.toLong() }
        transactions.removeAt(index)
    }
}